Sources/GRPC/GRPCClientConnection.swift

```diff
@@ -171,7 +153,163 @@ open class GRPCClientConnection {
   }
 }
 
+// MARK:- Configuration structures
+
+/// A target to connect to.
+public enum ConnectionTarget {
+  /// The host and port.
+  case hostAndPort(String, Int)
+  /// The path of a Unix domain socket.
+  case unixDomainSocket(String)
+  /// A NIO socket address.
+  case socketAddress(SocketAddress)
+
+  var host: String? {
+    guard case .hostAndPort(let host, _) = self else {
+      return nil
+    }
+    return host
+  }
+}
+
 extension GRPCClientConnection {
```

Comment:
Not really related to this PR, but do you think we should rename `GRPCClientConnection` to `ClientConnection`? Might be too generic, though.

Reply:
Great question, I've been wondering about this too since _most_ of the objects are prefixed with `GRPC`. We need a naming policy of some kind, ideally before tagging 1.0. I'm not sure what this should be though!

Reply:
Agreed, let's discuss this later!